### PR TITLE
Support quick adding of other output files

### LINKED ISSUE #184: Taskouput support quick adding other files
Currently suports viewing `stdout` and `stderr` which should be the most common output. However it would be nice to be able to quick view other files.

Add a + button to the list and a autocomplete with the list of files.

### PR DESCRIPTION
**Pull request by ascobie**
fix #184

### CODE REVIEW

File: test/app/components/task/details/output/task-log.component.spec.ts [hunk 1]
```diff
@@ -99,16 +99,17 @@ fdescribe("TaskDependenciesComponent", () => {
...
+            anyComponent = component as any;
             component.ngOnChanges({ jobId: component.jobId, task: component.task });
             fixture.detectChanges();
         });
 
-        // todo: do these need to be asyncy
-
         it("current task id set", () => {
-            const anyComponent = component as any;
+            anyComponent = component as any;
```
Comment (timotheeguerin) [inline test/app/components/task/details/output/task-log.component.spec.ts]: don't need this one

File: test/app/components/task/details/output/task-log.component.spec.ts [hunk 2]
```diff
@@ -123,96 +124,84 @@ fdescribe("TaskDependenciesComponent", () => {
...
-    //         expect(dependencies[0].dependsOn).toBe("1,2");
-    //         expect(dependencies[1].dependsOn).toBe("3 tasks");
-    //         expect(dependencies[2].dependsOn).toBe("no tasks");
-    //     });
-    // });
+    describe("loads additional task output files", () => {
+        let anyComponent: any;
+
+        beforeEach(() => {
+            anyComponent = component as any;
```
Comment (timotheeguerin) [inline test/app/components/task/details/output/task-log.component.spec.ts]: maybe move this to the top beforeEach to have it once only

File: app/components/task/details/output/task-log.component.ts [hunk 3]
```diff
@@ -1,58 +1,194 @@
...
+        return this.task.state !== TaskState.completed || !this.fileSizes[filename];
+    }
+
+    /**
+     * Ignore directories and any file that is either already in the map, or
+     * is one of the defaults.
+     */
+    private _canAddFileToMap(file: File) {
+        return !file.isDirectory &&
+            !this._fileMap[file.name] &&
```
Comment (timotheeguerin) [inline app/components/task/details/output/task-log.component.ts]: do !(file.name in this._fileMap) here

File: test/app/components/task/details/output/task-log.component.spec.ts (hunk 2 above)
Comment (ascobie) [inline test/app/components/task/details/output/task-log.component.spec.ts]: yeah i did think of that at the time, got distracted and forgot.

File: app/components/task/details/output/task-log.component.ts (hunk 3 above)
Comment (ascobie) [inline app/components/task/details/output/task-log.component.ts]: cool, will do.

**Pull request merged**